packages/frontend/src/components/CreateSwapButtons/ApproveButton.tsx

```diff
@@ -1,22 +1,28 @@
+import { useState } from 'react';
 import { showToast } from '@sifi/shared-ui';
 import { useWatch } from 'react-hook-form';
 import { useApprove } from 'src/hooks/useApprove';
 import { useAllowance } from 'src/hooks/useAllowance';
 import { SwapFormKey } from 'src/providers/SwapFormProvider';
+import { ApprovalModal } from 'src/modals';
 import { Button } from '../Button';
 
 const ApproveButton = () => {
-  const { refetch: refetchApproval, isFetching } = useApprove();
+  const [isApprovalModalOpen, setIsApprovalModalOpen] = useState(false);
+  const closeModal = () => setIsApprovalModalOpen(false);
```

Comment:
I think it would be cleaner to move the modal open/closing logic to the `useApprove` hook, this way we also don't need to pass the odd `closeModal` function to the hook as an argument. This would keep presentation and business logic better separated.